### ISSUE TITLE
Exception M 2.1.6: Area code is already set.

### ISSUE THREAD

**Issue by lfritsche** [Iazel/magento2-regenurl#13]
Regenerating Magento 2.1.6 Image cache by running command `bin/magento catalog:images:resize` leads to the following exception with this module enabled:

```
  [Magento\Framework\Exception\LocalizedException]  
  Area code is already set                          
                                                    


catalog:images:resize
```

Simply deactivate `bin/magento module:disable Iazel_RegenProductUrl` and the resize prozess proceeds.

**Comment by nkajic** [comment]
Confirming

**Comment by aaronware** [comment]
Some insight into the issue. https://github.com/magento/magento2/issues/5285 Looks like it's due to the setAreaCode being used within the constructor of the module

**Comment by nkajic** [comment]
Yup, works without setAreaCode('adminhtml') and I don't see any point for it since it is always run from CLI

**Comment by grafikchaos** [comment]
See PR #15 for fixed version

**Comment by peterjaap** [comment]
This is fixed in my fork, amongst other issues https://github.com/peterjaap/magento2-regenurl
